### DISCUSSION TITLE
SSAO and blending modes

### DISCUSSION THREAD

**Discussion by tottero** [google/filament#5037] [Q&A] [answered]
I'm trying to use SSAO on an object that has a material with blending mode. It looks like SSAO only gets computed for opaque or masked blending modes but not for others. I tried extending the conditions in ambient_occlusion.fs and also enabled depth write in the material but there is still no occlusion effect on the object.

Please advise what changes would be required on top of these to unlock SSAO support for custom materials with blending modes.

**Comment by pixelflinger** [comment, marked as answer]
Yes for the purpose of SSAO, translucent objects are ignored. There is no way to workaround that at the moment.

If you can modify the source code, you could try to change the definition of `SSAO` in `RenderPass.h` around line 180. Just remove the `DEPTH_FILTER_TRANSLUCENT_OBJECTS` -- it might work.

**Reply by tottero** [reply]
@pixelflinger  Thanks for the tip, it worked!
